src/lib/builders/menu/create.ts

```diff
@@ -361,17 +362,19 @@ export function createMenuBuilder(opts: _MenuBuilderOptions) {
 				}
 			}
 
-			const unsubPortal = effect([portal], ([$portal]) => {
-				if ($portal === null) return noop;
+			const unsubDerived = effect([portal], ([$portal]) => {
+				unsubPortal();
+				if ($portal === null) return;
 				const portalDestination = getPortalDestination(node, $portal);
-				if (portalDestination === null) return noop;
-				return usePortal(node, portalDestination).destroy;
```

Comment:
callbacks that are returned are called before a new effect runs, so I don't think this is necessary

Reply:
@TGlide I didn't realize that! So yes, this is unnecessary